node/server.js

```diff
@@ -2020,23 +2020,27 @@ function drop_something(player, monster, share) {
 	// if(player.level<50 && monster.type=="goo" && mode.low49_200xgoo) monster_mult=200;
 	if (D.drops.monsters[monster.type] && player.tskin != "konami") {
 		D.drops.monsters[monster.type].forEach(function (item) {
-			if (
-				((!monster.temp || item[0] > 0.00001) &&
-					Math.random() / share / player.luckm / monster.level / monster_mult < item[0]) ||
-				mode.drop_all
-			) {
+			// temp monsters should also roll for drops [7/17/24]
+			// 1) calculate drop modifier
+			// 2) use drop modifier on a random roll (0 <= N < 1) for 'actual drop rate'
+			// 3) drop item if the calculated result is under the drop rate's threshold
+			let dropModifier = share / player.luckm / monster.level / monster_mult;
+			let itemShouldDrop = (Math.random() / dropModifier) < item[0];
```

Comment:
```suggestion
			let itemShouldDrop = Math.random() / dropModifier < item[0];
```
This is the change Eslint wants. (The parenthesis are not necessary)

Reply:
I know what it wants and I agree they aren't _necessary_ but I believe they add considerable benefit in terms of readability.

To expand, I learned/was raised that extra parenthesis is added clarity, so it seems really arbitrary whether they exist or not. I can make the change if I need to, but I really don't think this should be an error in linting as if a JSON looked like:

```json
{
  'key': 'value',
  'key2': 'val,
}
```

And it's not super redundant like:
` (((randomNumber/dropModifier)))`

I know es-lint knows the proper order of operations but the parens adds clarity to have them there. Simply removing the parens will fix the issue, yes, but sometimes you want the parens for added clarity because someone who's not so saavy could look at that and, not so unreasonably, be like:

`Math.random() / dropModifier < item[0];`
Random number divided by modifier less than item[0] or something

Also, as an example, the following two are functionally the same but one will produce an error:

`a || (b && c)` will produce an error through es-lint
`a || b && c `   will not produce an error

I just don't think es-lint's, at least what I would consider to be, suggestion here is deserving of an error and I don't think every output coming out of es-lint is bible either.

So, the way I see it, I've got three options I can think of off the top of my head to address this:
1) Remove the parens
2) Add `/* eslint-disable */` as a comment above the expression and `/* eslint-enable */` below the line so it doesn't give an error
3) Update the .eslintrc to add `'no-extra-parens': off` to prevent the linter from even reporting on this pattern
4) Update the .eslintrc to add `'no-extra-parens': warn` to change the rule to a _warning_ instead of an error so it doesn't prevent the build

I am preferential to #2 or 4 because it is the least intrusive.

And, as added context the following are the current rules:
```
"rules": {
    "no-constant-condition": "off",
    "no-dupe-else-if": "off",
    "indent": "off",
    "no-empty": "warn",
    "no-inner-declarations": "warn",
    "no-unreachable": "warn",
    "no-undef": "warn",
    "no-unused-vars": "warn",
    "no-redeclare": "warn",
    "curly": "error",
    "prettier/prettier": "error",
    "one-var": ["error", "never"],
    "no-sequences": "error"
  }
 ```
 
So, we warn on when a variable is undefined but we error when there are extra (clarity) parens? It doesn't seem consistent to me.

If ultimately you want me to remove the parens, okay then accept the commit suggestion. It's not a hill I need to die on. But I still gotta walk up the metaphorical hill and push back a little because I feel this is superfluous and eslint being doctrinaire by setting the default to `all` unless you explicitly disable that as per option #3 or set it to a warning as per option #4.

Reply:
Eslint is the source of truth in our case because it ensures consistency across the whole codebase without the need for arbitrary decisions to be made.  
In your specific case, you can use an intermediary variable to keep your point of having clearer separation.  
I personally don't think the parenthesis add clarity because the expression is pretty self-explanatory.

Please refer to https://github.com/kaansoral/adventureland/pull/54 to get a better understanding of the way ESLint was configured. This rule will stay as an error because it is not currently triggering across the codebase like the current warnings do. Adding eslint to an existing codebase is always tricky and this is the process we've put in place:
- Use warnings for things we need to fix but there currently are too much to fix at once.
- The warnings will become errors once enough of them have been fixed in the existing code.